### ISSUE TITLE
SYNTAX ERROR FOR RETURN {'raw': statement 

### ISSUE THREAD

**Issue by perrycannon** [google/google-api-python-client#454]
I am receiving this error
   return {'raw': base64.urlsafe_b64encode(message.as_string().encode('UTF-8')).decode('ascii')}
         ^
SyntaxError: invalid syntax

I have used different versions of the return statement to return the message with the same result.
[qsEmailTest.txt](https://github.com/google/google-api-python-client/files/1469313/qsEmailTest.txt)

Thank you for helping
Perry

**Comment by bantini** [comment]
@perrycannon Can you share the code that you are using that is causing this error?

**Comment by perrycannon** [comment]
Sorry for the delay. I will post my code.
 from __future__ import print_functionimport httplib2import os
from apiclient import discoveryfrom apiclient import errorsfrom oauth2client import clientfrom oauth2client import toolsfrom oauth2client.file import Storage
import base64from email.mime.text import MIMEText
try:    import argparse    flags = argparse.ArgumentParser(parents=[tools.argparser]).parse_args()except ImportError:    flags = None
# If modifying these scopes, delete your previously saved credentials# at ~/.credentials/gmail-python-quickstart.jsonSCOPES = 'https://www.googleapis.com/auth/gmail.compose'CLIENT_SECRET_FILE = 'client_secret.json'APPLICATION_NAME = 'Gmail API Python Quickstart'

def get_credentials():    """Gets valid user credentials from storage.
    If nothing has been stored, or if the stored credentials are invalid,    the OAuth2 flow is completed to obtain the new credentials.
    Returns:        Credentials, the obtained credential.    """    home_dir = os.path.expanduser('~')    credential_dir = os.path.join(home_dir, '.credentials')    if not os.path.exists(credential_dir):        os.makedirs(credential_dir)    credential_path = os.path.join(credential_dir,                                   'gmail-python-quickstart.json')
    store = Storage(credential_path)    credentials = store.get()    if not credentials or credentials.invalid:        flow = client.flow_from_clientsecrets(CLIENT_SECRET_FILE, SCOPES)        flow.user_agent = APPLICATION_NAME        if flags:            credentials = tools.run_flow(flow, store, flags)        else: # Needed only for compatibility with Python 2.6            credentials = tools.run(flow, store)        print('Storing credentials to ' + credential_path)    return credentials
def main():    """Shows basic usage of the Gmail API.
    Creates a Gmail API service object and outputs a list of label names    of the user's Gmail account.    """    credentials = get_credentials()    http = credentials.authorize(httplib2.Http())    service = discovery.build('gmail', 'v1', http=http)
    results = service.users().labels().list(userId='me').execute()    labels = results.get('labels', [])
    if not labels:        print('No labels found.')    else:      print('Labels:')      for label in labels:        print(label['name'])
def SendMessage(service, user_id, message):  """Send an email message.
  Args:    service: Authorized Gmail API service instance.    user_id: User's email address. The special value "me"    can be used to indicate the authenticated user.    message: Message to be sent.
  Returns:    Sent Message.  """  try:    message = (service.users().messages().send(userId=user_id, body=message)               .execute())    print ('Message Id: %s' % message['id'])    return message  except errors.HttpError as error:    print ('An error occurred: %s' % error)

def CreateMessage(sender, to, subject, message_text):  """Create a message for an email.
  Args:    sender: Email address of the sender.    to: Email address of the receiver.    subject: The subject of the email message.    message_text: The text of the email message.
  Returns:    An object containing a base64url encoded email object.  """  message = MIMEText(message_text)  message['to'] = to  message['from'] = sender  message['subject'] = subject  raw = base64.b64encode(bytes(str(message))  return {'raw': base64.urlsafe_b64encode(message.as_string().encode('UTF-8')).decode('ascii')}
def main( sender, recepient, subject, text_body ):    credentials = get_credentials()    http = credentials.authorize(httplib2.Http())    service = discovery.build('gmail', 'v1', http=http)    testMessage = CreateMessage( sender, recepient, subject, text_body )    SendMessage( service, sender, testMessage )

if __name__ == '__main__':     main( 'xxxxxxxx@gmail.com', 'xxxxxxxxxx@yahoo.com', 'Test mail subject', 'Test mail body' )

    On Friday, November 17, 2017, 11:22:33 AM EST, Nilayan Bhattacharya <notifications@github.com> wrote:  
 
 
@perrycannon Can you share the code that you are using that is causing this error?

**Comment by bantini** [comment]
@perrycannon I think you missed a closing parenthesis in the line:

```python
  raw = base64.b64encode(bytes(str(message))
```

**Comment by perrycannon** [comment]
Thank you for your help. I had to install httplib2, and apiclient. I had to correct the son file name for the credential.

Now I get this error

Traceback (most recent call last):
  File "/home/pi/qsEmailTest.py", line 126, in <module>
    main( 'perryjames.cannon@gmail.com', 'perrycannon@yahoo.com', 'Test mail subject', 'Test mail body' )
  File "/home/pi/qsEmailTest.py", line 121, in main
    testMessage = CreateMessage( sender, recepient, subject, text_body )
  File "/home/pi/qsEmailTest.py", line 114, in CreateMessage
     raw = base64.b64encode(bytes(str(message))
 TypeError: string argument without an encoding
